app/models/contact_topic.rb

```diff
@@ -11,6 +11,16 @@ class ContactTopic < ApplicationRecord
 
   scope :active, -> { where(active: true, soft_delete: false) }
 
+  scope :with_answers_in, ->(case_contacts_scope) do
+    # unscope order in case it collides with distinct
```

Comment:
good comment

Reply:
To clarify that I understand this correctly

this line `case_contact_ids = case_contacts_scope.unscope(:order).select(:id).distinct` 
1. unscopes order which comes from https://github.com/rubyforgood/casa/blob/ee62ca93422548d05cb8161209c6b8f9285a8c81/app/models/contact_topic_answer.rb#L7
2. selects id to be later used in `.where(case_contact: {id: ...})`
3. De-dupes

and 
```
ContactTopic                                  
  .joins(contact_topic_answers: :case_contact)
  .where(case_contact: {id: case_contact_ids}) 
```
inner joins ContactTopics to ContactTopicAnswers to the rest of the query but uses the `where()` to ensure only the contact topics and topic answers for the filtered case contacts are kept.